tools/schema_loader.cc

```diff
@@ -252,6 +254,13 @@ std::vector<schema_ptr> do_load_schemas(std::string_view schema_str) {
             dd_impl.unwrap(ks).metadata->add_user_type(p->create_type(db));
         } else if (auto p = dynamic_cast<cql3::statements::create_table_statement*>(statement)) {
             schemas.push_back(p->get_cf_meta_data(db));
+            // CDC tables use a custom partitioner, which is not reflected when
+            // dumping the schema to schema.cql, so we have to manually set it here.
+            if (cdc::is_log_name(schemas.back()->cf_name())) {
```

Comment:
One could in theory create a table manually with the name `X_scylla_cdc_log`.

I think a more reliable way would be to use `is_log_for_some_table` (it's already a function exposed in cdc/log.hh). It checks if the name is of the form `X_scylla_cdc_log`, and then if the table `X` has CDC enabled.

```cpp
bool is_log_for_some_table(const replica::database& db, const sstring& ks_name, const std::string_view& table_name);
```

Reply:
Note that this is a loader which loads schemas from a `schema.cql` file, which usually only contains just what is absolutely necessary to parse an sstable using this schema. In case of a CDC table, it won't contain the table the CDC is enabled on, so we cannot verify said table's existence and the state of the cdc options.

Reply:
Ok. Then it should be enough to document the assumption made by the loader (if the name of the table ends with `_scylla_cdc_log`, we assume it's a CDC table that uses the CDC partitioner), and hope nobody is "clever" enough to name their tables like this.